src/java.base/linux/classes/jdk/internal/platform/CgroupSubsystemFactory.java

```diff
@@ -128,6 +156,36 @@ static CgroupMetrics create() {
         return Optional.of(result);
     }
 
+    private static boolean isRelevantControllerMount(String line) {
+         Matcher lineMatcher = MOUNTINFO_PATTERN.matcher(line.trim());
+         if (lineMatcher.matches()) {
+             String mountPoint = lineMatcher.group(1);
+             String fsType = lineMatcher.group(2);
+             boolean relevantControllerFound = false;
+             if (fsType.equals("cgroup")) {
+                 String filename = Paths.get(mountPoint).getFileName().toString();
+                 for (String fn: filename.split(",")) {
+                     switch (fn) {
+                         case MEMORY_CTRL: // fall through
+                         case CPU_CTRL:
+                         case CPUSET_CTRL:
+                         case CPUACCT_CTRL:
+                         case BLKIO_CTRL:
+                             relevantControllerFound = true;
+                             break;
+                         default: break; // ignore not recognized controllers
+                     }
+                 }
+             } else if (fsType.equals("cgroup2")) {
+                 relevantControllerFound = true;
+             }
+             return relevantControllerFound;
+         } else {
+           // fallback to old, pre JDK-8245543, behaviour
+           return line.contains("cgroup");
```

Comment:
Why are you doing this fallback rather than returning false??

Reply:
The idea was to have have some basics working even though the pattern doesn't match for some reason. I guess we can remove that as it's rather unlikely to not match.

Reply:
The change looks good.